core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1019,13 +1029,14 @@ class ReplicaManager(val config: KafkaConfig,
   }
 
   /**
-   * Append the messages to the local replica logs
+   * Append the messages to the local replica logs. ReplicaManager#appendRecords should usually be
+   * used instead of this method.
```

Comment:
Is there a strong reason why appendRecords is preferred, but in this case we don't use it?

Reply:
I did not use it here because it put stuff to the action queue and I don't want/need this in my case. Let me circle back on this to see if can do better. I wrote this a while ago and I admit that I went with the easiest solution back then.

Reply:
Those reasons make sense. The append path is pretty complex, so an overall refactor could do some good. But doesn't need to be a blocker here.

Reply:
I have refactored the implementation to use `appendRecords`. I think that it is better like this. Let me know what you think.

Reply:
lgtm :)